### DISCUSSION TITLE
i18n: Obtain default locale in path in middleware function

### DISCUSSION THREAD

**Discussion by bsanchez-aplyca** [vercel/next.js#39473] [Help]
Hello,

I'm trying to redirect (with 301) all routes from `/es/[*REST_OF-ROUTE*]` to `/[*REST_OF-ROUTE*]` just for the default locale (es) for SEO purposes, however, in the request the locale in path is removed as you can see in the image below.

![2022-08-10-123048_1914x1030_scrot](https://user-images.githubusercontent.com/92047731/183979818-01ae55df-3ea7-4dd7-9f45-fdd7b5ce9ca1.png)

So, is there any way to obtain the real pathname in the middleware function?

**Node's version:** 16
**NextJs's Version:** 12.1
**React's version:** 17.0.2

If you need more information, let me know. 

Thanks in advance!.

**Comment by izoukhai** [comment]
Same thing here with the latest version of Next, any news around here ? 
Seems like no one wants to answer this specific question, and there are literally 0 documentation about it

**Comment by izoukhai** [comment]
@bsanchez-aplyca Well it seems like it's not possible at all. There's a workaround though: https://www.izoukhai.com/blog/implement-internationalization-i18n-in-next-13-actually-working (using [next-translate](https://github.com/aralroca/next-translate))
